src/include/parser/scan_source.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "common/copy_constructors.h"
+#include "parser/statement.h"
+
+namespace kuzu {
+namespace parser {
+
+struct ScanSource {
+    std::string objectName;
+    std::vector<std::string> filePaths;
+    std::unique_ptr<Statement> statement;
+
+    ScanSource() = default;
+    DELETE_COPY_AND_MOVE(ScanSource);
+
+    inline bool fromObject() const { return !objectName.empty(); }
```

Comment:
Should we make ScanSource a hierachy  structure?
We can have BaseScanSource, and FileScanSource, statementScanSource.
Then you probably don't have to implement the fromObject(), fromFiles(), fromStatement() methods.